src/main/java/com/snowflake/kafka/connector/internal/IngestSdkProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal;
+
+import com.snowflake.kafka.connector.internal.streaming.StreamingUtils;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+public class IngestSdkProvider {
```

Comment:
Not a big fan of this class name, could this be `StreamingIngestClientGenerator`?

Reply:
I agree, it doesnt make sense in this PR. I want to expand it to manage all ingest sdk stuff though - so we can reduce the size of TopicPartitionChannel and make mocking ingest methods easier for testing. 

I can always change the name for now, but I am hoping to expand the scope of this. What do you think?

Reply:
Switched to ClientManager, resolving this issue